packages/create-cloudflare/templates-experimental/next/c3.ts

```diff
@@ -27,6 +32,14 @@ const configure = async () => {
 		startText: "Adding the Cloudflare adapter",
 		doneText: `${brandColor(`installed`)} ${dim(packages.join(", "))}`,
 	});
+
+	const { npm } = detectPackageManager();
+	await runCommand([npm, "run", NPM_TYPE_GEN_SCRIPT], {
```

Comment:
I feel like a simpler change that doesn't require changes to how C3 hooks are run, is just to put the command that generates the types into a variable. Then run it here and use it to update the package.json below.

E.g.

```suggestion
  const GENERATE_TYPES = ["wrangler", "types", "--env-interface", "CloudflareEnv", "cloudflare-env.d.ts"];
	await runCommand([npx, ...GENERATE_TYPES], {
```

Reply:
That would work too but I prefer using the actual script.

Do you think that calling `updatePackageScripts()` before `configure` might ever cause issues?

Reply:
I don't see the benefit in making this change to how C3 hooks run when a simple change to the c3.ts file would suffice. The update to the package.json is simply to make user facing scripts ready to go, not to support internal scripts.

Reply:
1) What I mentionned

https://github.com/cloudflare/workers-sdk/blob/df0e5bef817c09754471ac4842531a1e14e5576a/packages/create-cloudflare/src/frameworks/index.ts#L31-L35

> The update to the package.json is simply to make user facing scripts ready to go, not to support internal scripts.

the `.d.ts` is supposed to be generated/updated by running the "user facing script", that is what this PR implements.

Reply:
That comment is not relevant here, I believe. We are not running the CLI generator script in this command, which has to be run outside of a node package.
Here we are just running `wrangler types` command inside the package, which works fine with yarn.